utils/update.py

```diff
@@ -59,16 +59,16 @@ def update(o, **kwargs):
 
         with update(myrecord):
            myrecord.lastname = 'Smith'
-           myrecord.address.zipcode = '27707'
 
     """
-    cp = deepcopy(o)
+    cp = copy(o)
 
     # let the block presumably mutate o
-    yield
-
-    # swap the states of o and cp so that cp is the updated one
-    o.__dict__, cp.__dict__ = cp.__dict__, o.__dict__
+    try:
```

Comment:
I think this was done like this so that if it failed, the swap didn't occur. This was so that if the ManageIQ operation failed, then the internal reference wouldn't be manipulated as it would then be untrue. Can we preserve this functionality for now? Update will be totally replaced soon I feel anyway.